src/test/validatorTest.ts

```diff
@@ -141,6 +141,65 @@ describe('validator', () => {
+            });       
+            it("should error if second element is not a list or a function", () => {
+              const input = require('../../testData/invalid/json/5_invalid_intrinsic_select_2.json');
+              let result = validator.validateJsonObject(input);
+              expect(result).to.have.deep.property('templateValid', false);
+              expect(result['errors']['crit']).to.have.lengthOf(1);
+              expect(result['errors']['warn']).to.have.lengthOf(0);
+            });
+            it("should error if first element is not a number or does not parse to a number", () => {
+              const input = require('../../testData/invalid/json/5_invalid_intrinsic_select_3.json');
+              let result = validator.validateJsonObject(input);
+              expect(result).to.have.deep.property('templateValid', false);
+              expect(result['errors']['crit']).to.have.lengthOf(1);
+              expect(result['errors']['warn']).to.have.lengthOf(0);
+            });
+            it("should error if first element is not defined or is null", () => {
+              const input = require('../../testData/invalid/json/5_invalid_intrinsic_select_4.json');
+              let result = validator.validateJsonObject(input);
+              expect(result).to.have.deep.property('templateValid', false);
+              expect(result['errors']['crit']).to.have.lengthOf(1);
+              expect(result['errors']['warn']).to.have.lengthOf(0);
+            });
+            it("should error if only one element as argument list", () => {
+              const input = require('../../testData/invalid/json/5_invalid_intrinsic_select_5.json');
+              let result = validator.validateJsonObject(input);
+              expect(result).to.have.deep.property('templateValid', false);
+              expect(result['errors']['crit']).to.have.lengthOf(1);
+              expect(result['errors']['warn']).to.have.lengthOf(0);
+            });
+            it("should error if second element is null or undefined", () => {
+              const input = require('../../testData/invalid/json/5_invalid_intrinsic_select_6.json');
+              let result = validator.validateJsonObject(input);
+              expect(result).to.have.deep.property('templateValid', false);
+              expect(result['errors']['crit']).to.have.lengthOf(1);
+              expect(result['errors']['warn']).to.have.lengthOf(0);
+            });
+            it("should error if second element does not resolve to a list", () => {
+              const input = require('../../testData/invalid/json/5_invalid_intrinsic_select_7.json');
+              let result = validator.validateJsonObject(input);
+              expect(result).to.have.deep.property('templateValid', false);
+              expect(result['errors']['crit']).to.have.lengthOf(1);
+              expect(result['errors']['warn']).to.have.lengthOf(0);
+            });
```

Comment:
It would also be good to ensure the YAML shorthand is working as expected.
```
AvailabilityZone: !Select 
  - 0
  - !GetAZs 
    Ref: 'AWS::Region'

AvailabilityZone: !Select 
  - 0
  - Fn::GetAZs: !Ref 'AWS::Region'
```

Reply:
Yep